Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp

```diff
@@ -3759,17 +3759,14 @@ void BytecodeGenerator::emitCallDefineProperty(RegisterID* newObj, RegisterID* p
     }
 }
 
-RegisterID* BytecodeGenerator::emitReturn(RegisterID* src, ReturnFrom from)
+RegisterID* BytecodeGenerator::emitReturn(RegisterID* src)
 {
     // Normal functions and naked constructors do not handle `return` specially.
     if (isConstructor() && constructorKind() != ConstructorKind::Naked) {
         bool isDerived = constructorKind() == ConstructorKind::Extends;
         bool srcIsThis = src->index() == m_thisRegister.index();
 
-        if (isDerived && (srcIsThis || from == ReturnFrom::Finally))
-            emitTDZCheck(src);
```

Comment:
For case `emitReturn(context.completionValueRegister(), ReturnFrom::Finally);`, then `from == ReturnFrom::Finally` is true and `srcIsThis` is false. Then, previously we `emitTDZCheck` when `isDerived` is true. Are we sure in that case the `isDerived` is always false? So, `emitTDZCheck` is not needed.

Reply:
We are safe not to emit TDZ check in this case because the only reason we needed to check TDZ is because of automatically inserted `return`, which now performs the check on it's own due to `ensureThis()`.

In changelog, this is covered in 1).

Please note TDZ checks are only for uninitialized `this` value here; for userland code like `return foo; let foo;`, a TDZ error will be thrown by `ResolveNode<foo>`.